source/geod24/LocalRest.d

```diff
@@ -776,7 +734,10 @@ public final class RemoteAPI (API, alias S = VibeJSONSerializer!()) : API
         public void restart (void function (API) callback = null)
             @trusted
         {
-            C.send(this.childTid, ShutdownCommand!API(callback, true));
+            if (isMainThread())
+                scheduler.start({ this.listen_chn.write(Variant(ShutdownCommand!API(callback, true))); });
+            else
+                this.listen_chn.write(Variant(ShutdownCommand!API(callback, true)));
```

Comment:
I see many usages of this pattern... Why didn't you simply write a private method in the control interface which accepts a `Variant` and abstract this away ?